Dockerfile

```
@@ -1,10 +1,24 @@
-# Does NOT install Zig, so `cargo build --release` will fail if any crate
-# pulls in spawn-core-sys (Zig shim). Skip this image from CI builds or
-# install Zig 0.14.1 from https://ziglang.org/download/0.14.1/ before use.
+# Dockerfile — see ADR-010 for hardening roadmap.
+# Smoke image to verify Container Scan gate.
+# Uses .dockerignore to exclude sensitive data (addresses SonarCloud CRITICAL).
+# Installs Zig 0.14.1 in the builder stage so spawn-core-sys (used by some
+# workspace crates) can compile its Zig shim.
 FROM rust:1.85-slim AS builder
+# Install Zig 0.14.1 (pinned) — required by spawn-core-sys build script.
+ARG ZIG_VERSION=0.14.1
+RUN apt-get update \
```

Hadolint (2.15.1)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)